man-page/uu_nl.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH uu_nl 1  "uu_nl 0.0.27" 
.SH NAME
uu_nl \- Number lines of files
.SH SYNOPSIS
\fBuu_nl\fR [\fB\-\-help\fR] [\fB\-b\fR|\fB\-\-body\-numbering\fR] [\fB\-d\fR|\fB\-\-section\-delimiter\fR] [\fB\-f\fR|\fB\-\-footer\-numbering\fR] [\fB\-h\fR|\fB\-\-header\-numbering\fR] [\fB\-i\fR|\fB\-\-line\-increment\fR] [\fB\-l\fR|\fB\-\-join\-blank\-lines\fR] [\fB\-n\fR|\fB\-\-number\-format\fR] [\fB\-p\fR|\fB\-\-no\-renumber\fR] [\fB\-s\fR|\fB\-\-number\-separator\fR] [\fB\-v\fR|\fB\-\-starting\-line\-number\fR] [\fB\-w\fR|\fB\-\-number\-width\fR] [\fB\-V\fR|\fB\-\-version\fR] [\fIfile\fR] 
.SH DESCRIPTION
Number lines of files
.SH OPTIONS
.TP
\fB\-\-help\fR
Print help information.
.TP
\fB\-b\fR, \fB\-\-body\-numbering\fR=\fISTYLE\fR
use STYLE for numbering body lines
.TP
\fB\-d\fR, \fB\-\-section\-delimiter\fR=\fICC\fR
use CC for separating logical pages
.TP
\fB\-f\fR, \fB\-\-footer\-numbering\fR=\fISTYLE\fR
use STYLE for numbering footer lines
.TP
\fB\-h\fR, \fB\-\-header\-numbering\fR=\fISTYLE\fR
use STYLE for numbering header lines
.TP
\fB\-i\fR, \fB\-\-line\-increment\fR=\fINUMBER\fR
line number increment at each line
.TP
\fB\-l\fR, \fB\-\-join\-blank\-lines\fR=\fINUMBER\fR
group of NUMBER empty lines counted as one
.TP
\fB\-n\fR, \fB\-\-number\-format\fR=\fIFORMAT\fR
insert line numbers according to FORMAT
.br

.br
[\fIpossible values: \fRln, rn, rz]
.TP
\fB\-p\fR, \fB\-\-no\-renumber\fR
do not reset line numbers at logical pages
.TP
\fB\-s\fR, \fB\-\-number\-separator\fR=\fISTRING\fR
add STRING after (possible) line number
.TP
\fB\-v\fR, \fB\-\-starting\-line\-number\fR=\fINUMBER\fR
first line number on each logical page
.TP
\fB\-w\fR, \fB\-\-number\-width\fR=\fINUMBER\fR
use NUMBER columns for line numbers
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version
.SH EXTRA
STYLE is one of:

* a     number all lines
* t     number only nonempty lines
* n     number no lines
* pBRE  number only lines that contain a match for the basic regular
        expression, BRE

FORMAT is one of:

* ln    left justified, no leading zeros
* rn    right justified, no leading zeros
* rz    right justified, leading zeros
.SH VERSION
v0.0.27


completion/uu_nl.bash
_uu_nl() {
    local i cur prev opts cmd
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="uu_nl"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        uu_nl)
            opts="-b -d -f -h -i -l -n -p -s -v -w -V --help --body-numbering --section-delimiter --footer-numbering --header-numbering --line-increment --join-blank-lines --number-format --no-renumber --number-separator --starting-line-number --number-width --version [file]..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --body-numbering)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -b)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --section-delimiter)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -d)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --footer-numbering)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -f)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --header-numbering)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -h)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --line-increment)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -i)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --join-blank-lines)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -l)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --number-format)
                    COMPREPLY=($(compgen -W "ln rn rz" -- "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -W "ln rn rz" -- "${cur}"))
                    return 0
                    ;;
                --number-separator)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --starting-line-number)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -v)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --number-width)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -w)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

complete -F _uu_nl -o nosort -o bashdefault -o default uu_nl
